CI/deploy-mudlet-for-windows.sh

```diff
@@ -0,0 +1,331 @@
+  echo "to the bitness you want to work on. You may do this once for each of them should"
+  echo "you wish to do both."
+  exit 2
+elif [ "${MSYSTEM}" = "MINGW32" ]; then
+  export BUILD_BITNESS="32"
+  export BUILDCOMPONENT="i686"
+elif [ "${MSYSTEM}" = "MINGW64" ]; then
+  export BUILD_BITNESS="64"
+  export BUILDCOMPONENT="x86_64"
+else
+  echo "This script is not set up to handle systems of type ${MSYSTEM}, only MINGW32 or"
+  echo "MINGW64 are currently supported. Please rerun this in a bash terminal of one"
+  echo "of those two types."
+  exit 2
+fi
+
+cd "$GITHUB_WORKSPACE" || exit 1
+
+PublicTestBuild=false
+# Check if GITHUB_REPO_TAG is "false"
+if [[ "$GITHUB_REPO_TAG" == "false" ]]; then
+  echo "=== GITHUB_REPO_TAG is FALSE ==="
+
+  # Check if this is a scheduled build
+  if [[ "$GITHUB_SCHEDULED_BUILD" == "true" ]]; then
+    echo "=== GITHUB_SCHEDULED_BUILD is TRUE, this is a PTB ==="
+    MUDLET_VERSION_BUILD="-ptb"
+    PublicTestBuild=true
+  else
+    MUDLET_VERSION_BUILD="-testing"
+  fi
+
+  # Check if this is a pull request
+  if [[ -n "$GITHUB_PULL_REQUEST_NUMBER" ]]; then
+    # Use the specific commit SHA from the pull request head, since GitHub Actions merges the PR
+    BUILD_COMMIT=$(git rev-parse --short "$GITHUB_PULL_REQUEST_HEAD_SHA")
+    MUDLET_VERSION_BUILD="$MUDLET_VERSION_BUILD-PR$GITHUB_PULL_REQUEST_NUMBER"
+  else
+    BUILD_COMMIT=$(git rev-parse --short HEAD)
+
+    if [[ "$MUDLET_VERSION_BUILD" == "-ptb" ]]; then
+      # Get current date in YYYY-MM-DD format
+      DATE=$(date +%F)
+      MUDLET_VERSION_BUILD="$MUDLET_VERSION_BUILD-$DATE"
+    fi
+  fi
+fi
+
+# Convert to lowercase, not all systems deal with uppercase ASCII characters
+export MUDLET_VERSION_BUILD="${MUDLET_VERSION_BUILD,,}"
+export BUILD_COMMIT="${BUILD_COMMIT,,}"
+
+# Extract version from the mudlet.pro file
+VersionLine=$(grep "VERSION =" "$GITHUB_WORKSPACE/src/mudlet.pro")
+VersionRegex='= {1}(.+)$'
+
+# Use Bash regex matching to extract version
+if [[ $VersionLine =~ $VersionRegex ]]; then
+  VERSION="${BASH_REMATCH[1]}"
+fi
+
+# Check if MUDLET_VERSION_BUILD is empty and print accordingly
+if [[ -z "$MUDLET_VERSION_BUILD" ]]; then
+  # Possible release build
+  echo "BUILDING MUDLET $VERSION"
+else
+  # Include Git SHA1 in the build information
+  echo "BUILDING MUDLET $VERSION$MUDLET_VERSION_BUILD-$BUILD_COMMIT"
+fi
+
+# Check if we're building from the Mudlet/Mudlet repository and not a fork
+if [[ "$GITHUB_REPO_NAME" != "Mudlet/Mudlet" ]]; then
+  exit 2
+fi
+
+PACKAGE_DIR="$GITHUB_WORKSPACE/package-${MSYSTEM}-release"
+
+cd "$PACKAGE_DIR" || exit 1
+
+moveToUploadDir() {
+  local uploadFilename=$1
+  echo "=== Setting up upload directory ==="
+  local uploadDir="$GITHUB_WORKSPACE/upload"
+
+  # Check if the upload directory exists, if not, create it
+  if [[ ! -d "$uploadDir" ]]; then
+    mkdir -p "$uploadDir"
+  fi
+
+  echo "=== Moving files to upload directory ==="
+  mv "$PACKAGE_DIR/$uploadFilename" "$uploadDir/"
```

Comment:
Just a heads up, Github always zips artifacts, so it is double zipped right now.